### ISSUE TITLE
Wrong server (UMS) icon displayed on Samsung renderer

### ISSUE THREAD

**Issue by GoogleCodeExporter** [0xfab-ri/universal-media-server#40]
```
What steps will reproduce the problem?
1. Connect to UMS 2.3.0 using AllShare on a Samsung SmartTV EH5300 series

What is the expected output? What do you see instead?
I expected to see the UMS blue icon, but instead I saw some kind of default 
icon with a question mark on it. The UMS description is probably correct, but 
the icon isn't.

What version of the product are you using? On what operating system?
UMS 2.3.0; Win7 x64; Samsung UE40EH5300 with latest T-MST10PDEUC-1031.0 
firmware.
```

Original issue reported on code.google.com by `njasc...@gmail.com` on 1 Feb 2013 at 1:48

**Comment by GoogleCodeExporter** [comment]
```
I forgot to add something:
minidlna running on a router with the Tomato firmware corretly displays the 
"Tux" icon.
```

Original comment by `njasc...@gmail.com` on 1 Feb 2013 at 1:51

**Comment by GoogleCodeExporter** [comment]
```
Had the same issue on UE55ES8005
```

Original comment by `jakobste...@gmail.com` on 9 Apr 2013 at 4:50

**Comment by GoogleCodeExporter** [comment]
```
Same here LA40D550. Anyone looking to fix this? Seems some part of the DLNA 
spec is not up to scratch with UMS (or PMS)
```

Original comment by `tintin...@gmail.com` on 23 Apr 2013 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
I have the same on UE40ES5700 Samsung. My version works on WinXP.
```

Original comment by `Tomasz.Z...@gmail.com` on 17 Aug 2013 at 8:40

**Comment by GoogleCodeExporter** [comment]
```


I have done some network traces and found some possible causes of the problem. 
Attached are traces for the "description" request and response for UMS and 
Serviio. Following this response the TV does not even try to fetch the icon in 
the UMS case but immediately fetches the icon in the Serviio case. So these are 
the possible causes:

1. In UMS, the <URLBase> item value ends with a "/" character and the relative 
path for the icon starts with a "/" character. Surfing around I have seen 
several examples with and without the ending "/" but the relative path is 
always consistent with the choice. That is the "/" is either in the <URLBase> 
or in the relative path but not in both. So, I think this is a bug in UMS that 
most renderers forgive but maybe not Samsung. Notably, Serviio skips to provide 
the <URLBase> entirely.

2. In UMS, the <mimetype> item value is "image/jpeg" whereas the path name 
extension suggests it's a png. Serviio has <mimetype> "image/png" and this is 
one in the list that is fetched afterwards. Since the TV doesn't even fetch the 
icon in the UMS case, it might not accept jpeg as a format (and it doesn't 
consider the extension being png).

3. The size of the icon is 120*120 in the UMS case. Serviio delivers several 
icon options where the chosen one is 48*48.

It's difficult to know which one of those that is the actual cause of the 
problem, but I think nr 1 is a bug in UMS and nr 2 indicates an inconsistency 
in UMS. Providing a small icon option for 3 would at least not harm.
```

Original comment by `papikl...@gmail.com` on 24 Oct 2013 at 1:14

Attachments:
- [ws_ums_icon.txt](https://storage.googleapis.com/google-code-attachments/universal-media-server/issue-40/comment-5/ws_ums_icon.txt)
- [ws_serviio_icon.txt](https://storage.googleapis.com/google-code-attachments/universal-media-server/issue-40/comment-5/ws_serviio_icon.txt)

**Comment by GoogleCodeExporter** [comment]
```
Hi! It looks to me that it has been fixed in latest 3.0.0 (final) version. 
Great!
Thanks!
```

Original comment by `njasc...@gmail.com` on 3 Nov 2013 at 7:14

**Comment by GoogleCodeExporter** [comment]
```
Thanks for confirming this!
```

Original comment by `subjunk` on 4 Nov 2013 at 1:38
- Changed state: **Verified**
